test/middleware.spec.js

```diff
@@ -220,7 +220,7 @@ describe('Middleware', () => {
     });
   });
 
-  it('should return a promise that reject to selectQuery result', () => {
+  it('should return a promise that reject to selectRequest result', () => {
```

Comment:
Possible mistake in description, should it be: `should return a promise that rejects the selectRequest's result`?